executor/show.go

```diff
@@ -166,11 +166,30 @@ func (e *ShowExec) fetchShowEngines() error {
 	return nil
 }
 
+// move information_schema to the first, and the other are sorted in the origin ascending order.
+func moveInfoSchemaToFront(dbs []string) {
+	if strings.EqualFold(dbs[0], "INFORMATION_SCHEMA") {
+		return
+	}
+
+	for i := 1; i < len(dbs); i++ {
+		if strings.EqualFold(dbs[i], "INFORMATION_SCHEMA") {
+			tmp := dbs[i-1]
+			// the copy num is i-1, dbs[i-1] isn't copied to dbs[i]
+			copy(dbs[1:i], dbs[0:i-1])
+			dbs[i] = tmp
```

Comment:
ditto

Reply:
if just retain copy,
`	for i := 1; i < len(dbs); i++ {
		if strings.EqualFold(dbs[i], "INFORMATION_SCHEMA") {
			//tmp := dbs[i-1]
			// the copy num is i-1, dbs[i-1] isn't copied to dbs[i]
			copy(dbs[1:i], dbs[0:i-1])
			//dbs[i] = tmp
			dbs[0] = "INFORMATION_SCHEMA"
			return
		}
	}`
read the test case, run the test case
`cd executor && go test -check.f TestMoveInfoSchemaToFront`
 the error will be:
`
show_test.go:483:
    tk1.MustQuery("show databases").Check(testkit.Rows("INFORMATION_SCHEMA", "AAAA", "BBBB"))
/Users/andy/go/src/github.com/pingcap/tidb/util/testkit/testkit.go:55:
    res.c.Assert(resBuff.String(), check.Equals, needBuff.String(), res.comment)
... obtained string = "" +
...     "[INFORMATION_SCHEMA]\n" +
...     "[AAAA]\n" +
...     "[INFORMATION_SCHEMA]\n"
... expected string = "" +
...     "[INFORMATION_SCHEMA]\n" +
...     "[AAAA]\n" +
...     "[BBBB]\n"
... sql:show databases, args:[]

`